packages/split-button/src/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,140 @@
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const containerRef = useForwardedRef(ref, null);
+    const menuId = useIdAllocator({ prefix: 'lg-split-button-menu' });
+
+    const isAnchor = Component === 'a';
+
+    const buttonProps = {
+      // only add these props if not an anchor
+      ...(!isAnchor && { disabled, type }),
+    };
+
+    const sharedButtonProps = { variant, size, baseFontSize };
+
+    return (
+      <div className={cx(buttonContainerStyles, className)} ref={containerRef}>
+        <LeafyGreenProvider darkMode={darkMode}>
+          {/* TODO: remove when Button is updated to use `InferredPolymorphic` */}
+          {/* @ts-expect-error - Types of property `href` are incompatible. Button types href as string, but InferredPolymorphicProps types it as NodeUrlLike | ((string | NodeUrlLike) & string). This should not be an issue once Button is also using InferredPolymorphic. */}
+          <Button
+            as={Component}
+            {...sharedButtonProps}
+            {...buttonProps}
+            className={cx(buttonBaseStyles, {
+              [buttonThemeStyles(theme, variant)]: !disabled,
+            })}
+            {...rest}
+          >
+            {label}
+          </Button>
+          <Menu
+            {...sharedButtonProps}
+            spacing={spacing}
+            maxHeight={maxHeight}
+            adjustOnMutation={adjustOnMutation}
+            popoverZIndex={popoverZIndex}
+            usePortal={usePortal}
+            portalClassName={portalClassName}
+            portalContainer={portalContainer}
+            scrollContainer={scrollContainer}
+            align={align}
+            justify={justify}
+            containerRef={containerRef}
+            menuItems={menuItems}
+            id={menuId}
+            disabled={disabled}
+            open={open}
+            setOpen={setOpen}
+            onTriggerClick={onTriggerClick}
+            triggerAriaLabel={triggerAriaLabel}
+          />
+        </LeafyGreenProvider>
+      </div>
+    );
+  },
+  'SplitButton',
+);
+
+SplitButton.displayName = 'SplitButton';
+
+SplitButton.propTypes = {
+  darkMode: PropTypes.bool,
+  className: PropTypes.string,
+  align: PropTypes.oneOf(Object.values(Align)),
+  justify: PropTypes.oneOf(Object.values(Justify)),
+  variant: PropTypes.oneOf(Object.values(Variant)),
+  label: PropTypes.string.isRequired,
+  menuItems: PropTypes.element.isRequired,
+  baseFontSize: PropTypes.oneOf(Object.values(BaseFontSize)),
+  disabled: PropTypes.bool,
+  // Popover Props
+  popoverZIndex: PropTypes.number,
+  scrollContainer:
+    typeof window !== 'undefined'
+      ? PropTypes.instanceOf(Element)
+      : PropTypes.any,
+  portalContainer:
+    typeof window !== 'undefined'
+      ? PropTypes.instanceOf(Element)
+      : PropTypes.any,
+  portalClassName: PropTypes.string,
+} as any;
+
+// TODO: see if its better to use an array or menuItems
+// TODO: add test for menu closing
```

Comment:
General component comment - when the variant is `primary` or `danger` and the menu button is hovered, that box shadow seems pretty intrusive. Is that something design specifically wanted?